### PR TITLE
Clean up Either tests

### PR DESCRIPTION
**Pull request by ceedubs**
I think some of this code must have been written before we had
law-checking infrastructure in place for Eq/Order.

### CODE REVIEW

File: tests/src/test/scala/cats/tests/EitherTests.scala [hunk 1]
```diff
@@ -17,6 +18,11 @@ class EitherTests extends CatsSuite {
   val monad = implicitly[Monad[Either[Int, ?]]]
   val show = implicitly[Show[Either[Int, String]]]
 
+  val orderLaws = OrderLaws[Either[Int, String]]
+  checkAll("Either[Int, String]", orderLaws.eqv)
+  checkAll("Either[Int, String]", orderLaws.partialOrder(partialOrder))
+  checkAll("Either[Int, String]", orderLaws.order(order))
```
Comment (adelbertc) [inline tests/src/test/scala/cats/tests/EitherTests.scala]: Doesn't `order` include both `partialOrder` and `eqv` ?

Comment (ceedubs) [inline tests/src/test/scala/cats/tests/EitherTests.scala]: Please see comment by one @adelbertc: https://github.com/non/cats/blob/master/tests/src/test/scala/cats/tests/ConstTests.scala#L27-L30

;)

This probably isn't quite as good as the `Const` approach of using types that _only_ have the instance for the laws being checked.

Comment (adelbertc) [inline tests/src/test/scala/cats/tests/EitherTests.scala]: Ahh I see :-) I think I actually prefer your way over figuring our which types happen to have the instances I want

**Comment by adelbertc** [comment]
:+1:

**Comment by non** [comment]
:+1:

**Pull request merged**